mne/report.py

```diff
@@ -1582,6 +1596,19 @@ def _render_raw(self, raw_fname):
             div_klass='raw', id=global_id, caption=caption, info=raw.info,
             meas_date=meas_date, n_eeg=n_eeg, n_grad=n_grad, n_mag=n_mag,
             eog=eog, ecg=ecg, tmin=tmin, tmax=tmax)
+
+        if isinstance(self._raw_psd, dict):
+            from matplotlib.backends.backend_agg import FigureCanvasAgg
+            n_ax = sum(kind in raw for kind in _data_types)
+            fig, axes = plt.subplots(n_ax, 1, figsize=(6, 1 + 1.5 * n_ax),
+                                     dpi=92)
+            FigureCanvasAgg(fig)
+            img = _fig_to_img(raw.plot_psd, self.image_format,
```

Comment:
you compute it on the entire data?

Reply:
I think it would actually be wisest to compute it on the whole data -- for example if there is a huge artefact at the end of the data, only checking the first minute would not catch it. Can't it be sped up in other ways?

Actually, computing psds never takes too long for me?

Reply:
> you compute it on the entire data?

By default it does what `raw.plot_psd` will do, and yes we changed the default to be the entire data a while ago. This is implicitly controllable via the `dict` option.

> Actually, computing psds never takes too long for me?

It can take 30+ sec on some files. You don't have enough channels and/or time points :)

Reply:
I don’t think 30s is long i guess.